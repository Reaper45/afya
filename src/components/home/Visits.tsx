import * as React from "react"

import styled from "lib/emotion";
import FilterItem from 'components/common/FilterItem';

const VisitsWrapper = styled("div")`
  background: #fff;
  border-radius: 3px;
  padding: 1em;
  background: #fff;
  height: 100%;
  > b {
    display: block;
    text-transform: uppercase;
    color: ${props => props.theme.colors.accent};
    opacity: .66;
    font-weight: bolder;
    font-size: .7rem;
  }
  button {
    width: 100%;
  }
`;

export type VisitsType = Array<{
  count: number;
  title: string;
  key: string;
}>

const Visits: React.FC<{activeVisit: string; setActiveVisit: (key: string) => void; visits: VisitsType }> = ({ activeVisit, setActiveVisit, visits }) => (
  <VisitsWrapper data-name="visit-item">
    <b>Visits</b>
    <br/>
    {
      visits.map(visit =>
      <FilterItem
        active={activeVisit === visit.key}
        onClick={() => setActiveVisit(visit.key)}
        {...visit} />
    )}
  </VisitsWrapper>
);

export default Visits;
