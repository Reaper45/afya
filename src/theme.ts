export interface ITheme {
  colors: {
    primary: string;
    dark: string;
    light: string;
    secondary: string;
    accent: string;
  };
}

export const theme: ITheme = {
  colors: {
    primary: "#5A67D8",
    dark: "#3C366B",
    light: "#A3BFFA",
    secondary: "#222222",
    accent: "A0AEC0"
  }
};
