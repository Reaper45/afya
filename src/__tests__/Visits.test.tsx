import * as React from "react";
import { shallow, mount } from "enzyme";
import { ThemeProvider } from "emotion-theming";

import { theme } from "theme";

import Visits from "components/home/Visits";

const hospitalVisits: Array<{
  count: number;
  title: string;
  key: string;
}> = [
  {count: 7, title: "Kiambiu", key: "1" },
  {count: 12, title: "Mkuru Kwa Ruben", key: "2" },
  {count: 26, title: "Mkuru Kwa Njenga", key: "3" },
  {count: 38, title: "Baba dogo", key: "4" },
  {count: 41, title: "Kosovo", key: "5" },
  {count: 24, title: "Mukuru Kayaba", key: "6" },
];
 
it("renders without crashing", () => {
  const component = shallow(<Visits visits={hospitalVisits} />);
  expect(component).toMatchSnapshot();
})
  
it("renders all visits", () => {
  const component = mount(
    <ThemeProvider theme={theme}>
      <Visits visits={hospitalVisits} />
    </ThemeProvider>);
  expect(component.find("button")).toHaveLength(6);
})
 