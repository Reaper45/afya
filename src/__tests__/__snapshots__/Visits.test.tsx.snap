// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all visits 1`] = `
<Styled(div)>
  <b>
    Visits
  </b>
  <br />
  <FilterItem
    active={false}
    count={7}
    key="1"
    onClick={[Function]}
    title="Kiambiu"
  />
  <FilterItem
    active={false}
    count={12}
    key="2"
    onClick={[Function]}
    title="Mkuru Kwa Ruben"
  />
  <FilterItem
    active={false}
    count={26}
    key="3"
    onClick={[Function]}
    title="Mkuru Kwa Njenga"
  />
  <FilterItem
    active={false}
    count={38}
    key="4"
    onClick={[Function]}
    title="Baba dogo"
  />
  <FilterItem
    active={false}
    count={41}
    key="5"
    onClick={[Function]}
    title="Kosovo"
  />
  <FilterItem
    active={false}
    count={24}
    key="6"
    onClick={[Function]}
    title="Mukuru Kayaba"
  />
</Styled(div)>
`;

exports[`renders without crashing 1`] = `
<Styled(div)
  data-name="visit-item"
>
  <b>
    Visits
  </b>
  <br />
  <FilterItem
    active={false}
    count={7}
    key="1"
    onClick={[Function]}
    title="Kiambiu"
  />
  <FilterItem
    active={false}
    count={12}
    key="2"
    onClick={[Function]}
    title="Mkuru Kwa Ruben"
  />
  <FilterItem
    active={false}
    count={26}
    key="3"
    onClick={[Function]}
    title="Mkuru Kwa Njenga"
  />
  <FilterItem
    active={false}
    count={38}
    key="4"
    onClick={[Function]}
    title="Baba dogo"
  />
  <FilterItem
    active={false}
    count={41}
    key="5"
    onClick={[Function]}
    title="Kosovo"
  />
  <FilterItem
    active={false}
    count={24}
    key="6"
    onClick={[Function]}
    title="Mukuru Kayaba"
  />
</Styled(div)>
`;
